### PR TITLE
Feat/feide login

### PR DESCRIPTION
**Pull request by theodorklauritzen**

### CODE REVIEW

File: src/actions/users/update.ts [hunk 1]
```diff
@@ -27,6 +27,8 @@ export async function updateUser(id: number, rawdata: FormData | UpdateUserSchem
     }
 }
 
+// These function should maybe be in another place than server actions? @Paulijuz
```
Comment (Paulijuz) [inline src/actions/users/update.ts:30]: this comment should be removed :)

File: src/prisma/schema.prisma [hunk 2]
```diff
@@ -50,6 +59,16 @@ model Credentials {
   @@map("Authentication")
 }
 
+model FeideAccount {
+  id          String   @id
+  accessToken String   @db.Text
+  email       String?  @unique
```
Comment (Paulijuz) [inline src/prisma/schema.prisma:65]: should maybe not be optional? i think it's possible to fix

**Review by Paulijuz — CHANGES_REQUESTED** [review]
Looks great, just needs some small changes.

File: src/actions/users/update.ts (hunk 1 above)
Comment (theodorklauritzen) [reply src/actions/users/update.ts:30]: This is still a problem, will be fixed in another PR

File: src/app/(auth)/register/page.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    const sexOptions = [
+        { value: 'FEMALE', label: 'Kvinne' },
+        { value: 'MALE', label: 'Mann' },
+        { value: 'OTHER', label: 'Annet' },
+    ]
+
+    const lastUsername = userAuth.user?.username
+    let lastPassword: string = ''
+
+    async function callback() {
```
Comment (JohanHjelsethStorstad) [inline src/app/(auth)/register/page.tsx:42]: Arrow function

**Review by JohanHjelsethStorstad — CHANGES_REQUESTED** [review]

File: src/actions/users/schema.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { zfd } from 'zod-form-data'
 const userSchema = z.object({
     username: z.string().max(50).min(2),
     password: z.string().max(50).min(2),
+    sex: z.enum(['MALE', 'FEMALE', 'OTHER']),
```
Comment (Paulijuz) [inline src/actions/users/schema.ts:7]: Use built in zod native enum type

File: src/app/(auth)/register/page.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    return <Form
+        title="Registrer bruker"
+        submitText="Registrer bruker"
+        action={registerUser}
+        successCallback={() => signIn('credentials', {
+            username: lastUsername,
+            password: lastPassword,
+            redirect: true,
+            callbackUrl
+        })}
```
Comment (Paulijuz) [inline src/app/(auth)/register/page.tsx:51]: Sjekk ut om du trenger å fjerne http

**Review by Paulijuz — CHANGES_REQUESTED** [review]
Looks good, just needs some small changes

**Review by Paulijuz — APPROVED** [review]
Looks good. Have not tested it, but i've seen it works so it's probably fine. :)

File: src/auth/feide/FeideProvider.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
+import 'server-only'
+import { readJWTPayload } from '.'
+import type { Awaitable } from 'next-auth'
+import type { Provider } from 'next-auth/providers/index'
+import type { FeideGroup, ExtendedFeideUser, AdapterUserCustom } from './Types'
+
+export type PropType = {
```
Comment (JohanHjelsethStorstad) [inline src/auth/feide/FeideProvider.ts:7]: Rename

File: src/actions/studyprogrammes/create.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+                }
+            },
+            select: returnSelections
+        })
+
+        const existingStudyCodes = new Set(exists.map(program => program.code))
+        const newStudyPrograms = programs.filter(program => !existingStudyCodes.has(program.code))
+        let createdStudyPrograms: Array<{studyProgram: PrismaResults}> = []
+
+        if (newStudyPrograms.length > 0) {
```
Comment (JohanHjelsethStorstad) [inline src/actions/studyprogrammes/create.ts:68]: just studyProgram.length?

File: src/app/(auth)/register/page.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    const router = useRouter()
+
+    const userAuth = useUser()
+    if (userAuth.status !== 'authenticated') {
+        router.push('/login')
+    }
+
+    if (userAuth.user?.acceptedTerms) {
+        router.push('/users/me')
```
Comment (JohanHjelsethStorstad) [inline src/app/(auth)/register/page.tsx:23]: prefer const { push } = useRouter()

File: src/app/components/UI/Select.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import styles from './Select.module.scss'
+import { v4 } from 'uuid'
+
+export type Proptypes = {
+    name: string,
+    label?: string,
+    options: Array<{
```
Comment (JohanHjelsethStorstad) [inline src/app/components/UI/Select.tsx:7]: prefer [] over generic Array

File: src/app/users/[username]/page.tsx [hunk 10]
```diff
@@ -18,6 +18,7 @@ export default async function User({ params }: PropTypes) {
     const me = params.username === 'me'
     const username = me ? user.username : params.username
 
+    // REFACTOR
```
Comment (JohanHjelsethStorstad) [inline src/app/users/[username]/page.tsx:21]: // TODO: REFACTOR

**Review by JohanHjelsethStorstad — CHANGES_REQUESTED** [review]
Just change this

**Review by JohanHjelsethStorstad — CHANGES_REQUESTED** [review]
Just change this

**Review by Paulijuz — APPROVED** [review]
As a Theodor once said, looks good.

**Review by JohanHjelsethStorstad — APPROVED** [review]
looks good

**Pull request merged**